src/gtfs/read.rs

```diff
@@ -448,61 +472,147 @@ fn get_modes_from_gtfs(
-                geometry_id: None,
-                opening_time: None,
-                closing_time: None,
-            };
-            lines.push(l);
-        },
-        RouteReadType::RouteAsNtmRoute => {
-            // TODO Build lines from GTFS routes as routes
-            unimplemented!();
+
+    let line_code = |r: &Route| {
+        if r.short_name.is_empty() {
+            None
+        } else {
+            Some(r.short_name.to_string())
         }
+    };
+
+    let line_agency = |r: &Route| {
+        r.agency_id
+            .as_ref()
+            .map(|id| id.to_string())
+            .unwrap_or_else(default_agency_id)
+    };
+
+    for routes in map_line_routes.values() {
+        let r = get_route_with_smallest_name(routes);
+        lines.push(objects::Line {
+            id: r.id.to_string(),
+            code: line_code(r),
+            codes: KeysValues::default(),
+            object_properties: KeysValues::default(),
+            comment_links: CommentLinksT::default(),
+            name: r.long_name.to_string(),
+            forward_name: None,
+            forward_direction: None,
+            backward_name: None,
+            backward_direction: None,
+            color: r.color.clone(),
+            text_color: r.text_color.clone(),
+            sort_order: r.sort_order,
+            network_id: line_agency(r),
+            commercial_mode_id: r.route_type.to_gtfs_value(),
+            geometry_id: None,
+            opening_time: None,
+            closing_time: None,
+        });
     }
+
     lines
 }
 
-pub fn read_routes<P: AsRef<path::Path>>(path: P, collections: &mut Collections) {
-    let path = path.as_ref().join("routes.txt");
-    let mut rdr = csv::Reader::from_path(path).unwrap();
-    let gtfs_routes: Vec<Route> = rdr.deserialize().map(Result::unwrap).collect();
+fn make_routes(
+    gtfs_trips: &[Trip],
+    map_line_routes: &MapLineRoutes,
+) -> Result<Vec<objects::Route>> {
+    let mut routes = vec![];
+
+    let get_id = |r: &Route, d: &DirectionType| match *d {
+        DirectionType::Forward => r.id.clone(),
+        DirectionType::Backward => r.id.clone() + "_R",
+    };
+
+    let get_direction_name = |d: &DirectionType| match *d {
+        DirectionType::Forward => "forward".to_string(),
+        DirectionType::Backward => "backward".to_string(),
+    };
+
+    for rs in map_line_routes.values() {
+        let sr = get_route_with_smallest_name(rs);
+        for r in rs {
+            let mut route_directions: HashSet<&DirectionType> = HashSet::new();
+            for t in gtfs_trips.iter().filter(|t| t.route_id == r.id) {
+                route_directions.insert(&t.direction);
+            }
+            if route_directions.is_empty() {
+                ensure!(
+                    !route_directions.is_empty(),
+                    "Coudn't find trips for route_id {}",
+                    r.id,
+                );
+            }
```

Comment:
@prhod we fail if a route has no trip?

Reply:
If possible, please don't panic but add a warning  log.
Can you avoid creating empty Routes (and therefore empty lines) ?